### ISSUE TITLE
Provision for giving feedback right within Fluent Search

### ISSUE THREAD

**Issue by MakeshVineeth** [adirh3/Fluent-Search#230]
In order to cover more ground and to make giving feedback as much easy as possible for the users, it would be awesome if there is a way to give feedback right within FS. Also as you can see, right now, there is not much information within FS on how to send feedback i.e., there are no direct links for the same. People either have to go through GitHub website or through Discord for that. Not to mention, Tracking Discord messages can become quite a mess at times & also we don't have to make users create accounts on GitHub just to write few sentences.

I feel it is the right time for introducing a feedback system within FS and I have following suggestions for feedback window UI:

- A Text Box for **Title**. It should be **mandatory** field.
- A fairly big Text Box for typing the actual feedback text. It is **mandatory** field.
- A drop down menu to select whether the issue is related to **Bug** or **Suggestion**.
- A Text Box for mentioning Discord ID. **Optional** field.
- One more text box for Email. **Optional** field.
- A Checkbox that says **"Allow us to contact you to follow up with the issue"**. Enabled by default.
- Ability to attach upto 10 (more or less) images/videos to demonstrate the issue.
- FS version and Windows version should automatically be retrieved without needing for the user to type manually.
- Auto retrieve System Information like Processor, RAM etc, if needed. Let users **choose** whether to include this info while submitting the feedback.
- A **Send feedback** button in the Context Menu of Tray Icon and also in the top bar of Settings.
- Last but not least, **Submit** button at the bottom of feedback window. (obviously)

It doesn't have to be connected to GitHub Issues necessarily, just trying to keep suggestion as much simple as possible, so that it is easy for both users and also for dev to implement it. Rather, dev @adirh3 can have his own private database for storing these feedbacks and he may track it from there.

**Comment by Metroid-Prime** [comment]
I suggested something like this a while back and I think it's a good idea BUT it'll make tracking issues harder since it'll fragment feedback 3 primary ways (github, discord, & feedback form. I assume not much feedback by email)

**Comment by MakeshVineeth** [comment]
> I suggested something like this a while back and I think it's a good idea BUT it'll make tracking issues harder since it'll fragment feedback 3 primary ways (github, discord, & feedback form. I assume not much feedback by email) 

Yeah, tracking can be bit difficult. However from what I've seen there is so much feedback given on Discord channel itself lately, hence this issue was opened, having a built-in feedback will make things easier for the users.
